### ISSUE TITLE
Improve minutes handling

### ISSUE THREAD

**Issue by DirkHoffmann** [indico/indico#1827]
I would like to trigger some brainstorming about the minutes mechanism in Indico. I guess it is not very much used, but that is probably because the minutes management is not very convenient in nowadays context. I believe it could be easily upgraded to make the use of external tools obsolete.
* Minutes should have (minimal, maybe only last 3 versions) history of edits.
* Status flag (empty - editing - final) associated with colours (red-yel-grn) and locked when final. This is in particular useful for minutes that are compiled from sub-contribution minutes.
* With history comes the possibility to revert to a previous version ;-)
* Do minutes have to be compiled statically in the case of sub-contribution-minutes, or could they be compiled dynamically?
* Improve the style (use HTML tags) of compiled minutes.
* ...

**Comment by ThiefMaster** [comment]
I wonder if it even makes sense to keep text-based minutes as "material". To me, minutes look like data that would fit much better in a database table than the material file storage.

Additionally, minutes do have some special behavior (like being editable from within indico) while other material doesn't.

**Comment by DirkHoffmann** [comment]
Yes, they are quite different from "material", which has probably been chosen for anything else than the common metadata of all "conferences" (which in turn became conference+seminar+meeting). [Are you going to change conference :arrow_right: event during the big redesign? In last consequence, MaKaC should become MaKaE then :wink:.]

More seriously, here is an important other point:
* Extract list-of-actions from minutes. That is one of the basic functions that external tools provide for writing "good minutes". In addition to links and highlighting there could be a marker for "Action". Or a special syntax convention like the TODO: tags in Eclipse?

**Comment by DirkHoffmann** [comment]
NB: Once this becomes mature enough (and unless the central team objects), I think it should go through a "review" on the indico-admins list. So consider this as a ticket to write up reasonable specs for a better handling of the "minutes material".

**Comment by pferreir** [comment]
Thanks for the suggestions. We are currently reworking material/minute handling. Minutes will now work separately from materials, which will allow for easier future changes. They will be now called "notes" and there will be some interface improvements and, yes, we will save the history in the database.

The "restore" feature is future work. But with the database structures in place, it's a matter of doing some interface development.

**Comment by pferreir** [comment]
But, as you say, let's see what the community thinks. 2.0 will already be a step forward but the things that you mention could be very interesting.

**Comment by DirkHoffmann** [comment]
> Thanks for the suggestions. We are currently reworking material/minute
> handling. Minutes will now work separately from materials, which will allow
> for easier future changes.

Good! However it will add some workload on the developers to maintain a 
separate kind of "attachment" to be handled for an event.

> They will be now called "notes" and there will be some interface 
> improvements and, yes, we will save the history in the database.

Be careful with renaming. The term "minutes" indicates intuitively that people 
can use it for that purpose, whereas "notes" are in general a less qualitative 
record of a meeting. (Maybe that is on purpose? Minutes can of course be 
written offline with an external program, starting from "notes", and be 
attached to the event as material.)

If you redesign the minutes/note function, think about collaborative editing à 
la etherpad or linking to an etherpad instance anyway. The latter will save 
you a lot of work, but makes Indico dependent on availability / installation 
of an etherpad for a given Indico instance. (Should be optional anyway.)

Cheers!
 									Dirk

**Comment by pferreir** [comment]
> Good! However it will add some workload on the developers to maintain a
separate kind of "attachment" to be handled for an event.

I believe the way it worked was actually making it harder for us. Events had a specific material type called "minutes" and inside a single resource. This single use-case had to be handled separately, and it was easy for users to ruin the minutes by deleting the material type.

 > Be careful with renaming.

This is why we didn't change it at all on the interface: we want to know what people want to call it. One thing is for sure, people use this for more than "minutes". Minutes should be a summary of a meeting, after it happened. For instance, here in our team we write the "minutes" beforehand.

So, we're not sure about the naming. "Notes" seemed neutral enough, at least code-wise. If we want Indico to be a general-purpose event organization application, we have to find a term that is neutral enough and not misleading.

> collaborative editing à la etherpad or linking to an etherpad instance anyway

There were people interested in contributing to an Indico etherpad extension. However, there are still some challenges, e.g. the fact that etherpad doesn't have any serious security nor support for authorization protocols such as OAuth. That's pretty much a showstopper :disappointed:

**Comment by ThiefMaster** [comment]
What could be a really useful feature is a confirmation prompt if someone else edited the same notes, i.e. if you'd overwrite notes you haven't even seen.

**Comment by DirkHoffmann** [comment]
On Wed, 1 Jul 2015, Adrian wrote:

> What could be a really useful feature is a confirmation prompt if someone
> else edited the same notes, i.e. if you'd overwrite notes you haven't even
> seen.

I hope there are not so many people who co-edit minutes/notes for the same 
meeting. The most annoying in the present solution is that minutes can be 
*totally* lost without possibility to recover, in case the login token is 
timed out or the network connection is unreliable.

Of course there are many features like versioning, conflict warning, 
simultaneous editing à la etherpad, who seem so natural to us today, because 
other systems exist for that. I think the question is, what is reasonable for 
the central developers to add without being totally redundant with other 
tools.

Probably it is sufficient to make the present system more reliable and offer 
more sophisticated solutions as links to etherpad and the like. Also the 
possibility to compile minutes from several sessions/sub-elements is a good 
feature to keep.

Finally, "notes" ("online notes") may be a good choice, as it allows to keep 
the term "minutes" for a final document copying "notes" into OOo-Writer to 
make the final PDF.

More interesting than reinventing something too close to etherpad would 
probably be:
  * Possibility to "publish" notes, during writing or afterwards; by default 
they would be visible only to the editors or potential editors (contribution 
or session manager?).
  * Possibility to "lock" notes independently from the event itself (alike the 
event lock, with possibility to unlock, but you would know that they are in 
principle finished).

**Comment by DirkHoffmann** [comment]
> There were people interested in contributing to an Indico etherpad
> extension. However, there are still some challenges, e.g. the fact that
> etherpad doesn't have any serious security nor support for authorization
> protocols such as OAuth. That's pretty much a showstopper :disappointed:

Who wants to use etherpad knows that.

The authorisation is managed by the "hidden" hash in the URL, which cannot be 
guessed from outside. Of course any meeting participant would be able not only 
to see, but also to edit. It is a different (new) use case, I think. Etherpad 
allows to save/freeze a version, if you cannot trust your collaborators that 
they edit it silently.

On the other hand, there is probably not more needed to work with etherpad 
than the possibility to add an URL as material, which points to your session 
etherpad.
